irods/src/main/java/ch/cyberduck/core/irods/IRODSListService.java

```diff
@@ -50,34 +45,45 @@ public IRODSListService(IRODSSession session) {
     @Override
     public AttributedList<Path> list(final Path directory, final ListProgressListener listener) throws BackgroundException {
         try {
-            final AttributedList<Path> children = new AttributedList<Path>();
-            final IRODSFileSystemAO fs = session.getClient();
-            final IRODSFile f = fs.getIRODSFileFactory().instanceIRODSFile(directory.getAbsolute());
-            if(!f.exists()) {
-                throw new NotfoundException(directory.getAbsolute());
+            final IRODSConnection conn = session.getClient();
+
+            String logicalPath = directory.getAbsolute();
+            if(!IRODSFilesystem.exists(conn.getRcComm(), logicalPath)) {
+                throw new NotfoundException(logicalPath);
             }
-            for(File file : fs.getListInDirWithFileFilter(f, TrueFileFilter.TRUE)) {
-                final String normalized = PathNormalizer.normalize(file.getAbsolutePath(), true);
+
+            final AttributedList<Path> children = new AttributedList<Path>();
+
+            for(CollectionEntry entry : new IRODSCollectionIterator(conn.getRcComm(), logicalPath)) {
+                final String normalized = PathNormalizer.normalize(entry.path(), true);
                 if(StringUtils.equals(normalized, directory.getAbsolute())) {
                     continue;
                 }
-                final PathAttributes attributes = new PathAttributes();
-                final ObjStat stats = fs.getObjStat(file.getAbsolutePath());
-                attributes.setModificationDate(stats.getModifiedAt().getTime());
-                attributes.setCreationDate(stats.getCreatedAt().getTime());
-                attributes.setSize(stats.getObjSize());
-                attributes.setChecksum(Checksum.parse(Hex.encodeHexString(Base64.decodeBase64(stats.getChecksum()))));
-                attributes.setOwner(stats.getOwnerName());
-                attributes.setGroup(stats.getOwnerZone());
-                children.add(new Path(directory, PathNormalizer.name(normalized),
-                        file.isDirectory() ? EnumSet.of(Path.Type.directory) : EnumSet.of(Path.Type.file),
-                        attributes));
+
+                PathAttributes attrs = new PathAttributes();
+                attrs.setCreationDate(entry.createdAt() * 1000L);
+                attrs.setModificationDate(entry.modifiedAt() * 1000L);
+
+                EnumSet<Path.Type> type = EnumSet.of(Path.Type.file);
+
+                if(entry.isCollection()) {
+                    attrs.setDirectoryId(entry.id());
+                    type = EnumSet.of(Path.Type.directory);
+                }
+                else if(entry.isDataObject()) {
+                    attrs.setFileId(entry.id());
+                    attrs.setSize(entry.dataSize());
+                    attrs.setChecksum(IRODSChecksumUtils.toChecksum(entry.checksum()));
+                }
+
```

Comment:
Ideally conversion to `PathAttributes` can be moved to `AttributesAdapter` implementation.